# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/FontFaceSetWorkerImpl.cpp
# Commit: 7802e5f1bd13
# Full Hash: 7802e5f1bd1395f19f2d78636787c41f64192bd7
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P3 Remove WorkerRunnable::mWorkerPrivate. r=dom-worker-reviewers,asuth
#   
#   WorkerRunnable no longer keeps a raw pointer(mWorkerPrivate) for the associated WorkerPrivate in this patch.
#   Removing the WorkerRunnable::mWorkerPrivate needs to fix the following problems.
#   
# ==============================================================================

diff -r ffc69aa8f2c3 -r 7802e5f1bd13 layout/style/FontFaceSetWorkerImpl.cpp
--- a/layout/style/FontFaceSetWorkerImpl.cpp	Fri Apr 19 09:41:57 2024 +0000
+++ b/layout/style/FontFaceSetWorkerImpl.cpp	Fri Apr 19 09:41:58 2024 +0000
@@ -181,7 +181,7 @@
    public:
     FontFaceSetWorkerRunnable(WorkerPrivate* aWorkerPrivate,
                               std::function<void()>&& aFunc)
-        : WorkerThreadRunnable(aWorkerPrivate, "FontFaceSetWorkerRunnable"),
+        : WorkerThreadRunnable("FontFaceSetWorkerRunnable"),
           mFunc(std::move(aFunc)) {}
 
     bool WorkerRun(JSContext* aCx, WorkerPrivate* aWorkerPrivate) override {
@@ -195,7 +195,7 @@
 
   RefPtr<FontFaceSetWorkerRunnable> runnable =
       new FontFaceSetWorkerRunnable(workerPrivate, std::move(aFunc));
-  runnable->Dispatch();
+  runnable->Dispatch(workerPrivate);
 }
 
 uint64_t FontFaceSetWorkerImpl::GetInnerWindowID() {